src/Domain/LeanCode.DomainModels.MassTransitRelay/Testing/ResettableBusActivityMonitor.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MassTransit;
+using MassTransit.Testing.Indicators;
+using MassTransit.Util;
+
+namespace LeanCode.DomainModels.MassTransitRelay.Testing
+{
+    public class ResettableBusActivityMonitor :
+        IBusActivityMonitor,
+        IReceiveObserver,
+        IConsumeObserver,
+        IDisposable
+    {
+        private readonly object mutex = new object();
+        private readonly AsyncManualResetEvent inactive = new AsyncManualResetEvent(true);
+        private readonly RollingTimer timer;
+
+        private int consumersInFlight;
+        private int receiversInFlight;
```

Comment:
Since all writes to these variables are protected with a `lock` anyway, maybe you could mark them with `volatile` and get rid of usages of `Interlocked` and `Volatile` classes? Using both seems like an overkill, but I might be wrong here.

Reply:
Yep, you're right. I totally forgot about the keyword.